### DISCUSSION TITLE
🤔 input의 defaultValue는 왜 설정 하셨을까요?

### DISCUSSION THREAD

**Discussion by kimInDa** [FRONTENDSCHOOL6/react-project-11#60] [Q&A] [answered]
선생님께서 로그인과 회원가입 예제 만드신 코드를 보는데,
input에 defaultValue를 설정하신 것을 보았습니다.
딱히 기능이 있는 것 같지는 않은데 왜 사용하신걸까요..?

![2023-09-15_01-25-16 png](https://github.com/FRONTENDSCHOOL6/react-project-11/assets/105577805/62712bfe-272b-40b0-b7ca-0c37da228800)

**Comment by ryudg** [comment, marked as answer]
input 요소에 value 대신 defaultValue를 설정하는 주된 이유는 컴포넌트의 input을 "제어되지 않는uncontrolled" 컴포넌트로 만들기 위함입니다.

즉, input의 defaultValue가 아닌 value에 `formState.username`이 할당되면 input의 조작으로 값을 변경할 수 없고 useState의 set함수만이 만이 value를 변경할 수 있다는 의미입니다. 

참고 자료 : https://jihyundev.tistory.com/23

**Reply by kimInDa** [reply]
오우 이해되었습니다 감사합니다!

**Reply by summer787** [reply]
다인님이 설명해 주셨던 부분이 이 부분이었네요! 감사합니다!
